### ISSUE TITLE
typescript/explicit-module-boundary-types: diagnostic anchored on `export const` declaration line instead of the arrow function, so `eslint-disable-next-line` is ignored and reported unused

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25627]
## Summary

For multi-line `export const NAME = (...) => {}` declarations, oxlint's `typescript/explicit-module-boundary-types` anchors the "Missing return type on function" diagnostic on the `export const NAME =` declaration line, whereas `@typescript-eslint/explicit-module-boundary-types` anchors it on the arrow function expression (the `=>` token). As a result, an `// eslint-disable-next-line` placed on the line above the `) =>` — where ESLint reports — does not suppress oxlint's finding, and oxlint additionally reports that directive as unused. This silently breaks existing in-source suppressions during an ESLint→oxlint migration.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.ts`:

```ts
export const showLastActiveUsersWarningDialog = async (
	users: string[],
	organization: {id: string}
	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
) => {
	return {users, organization}
}
```

`.oxlintrc.json`:

```json
{
	"plugins": ["typescript"],
	"categories": {"correctness": "off"},
	"rules": {
		"typescript/explicit-module-boundary-types": ["error", {"allowArgumentsExplicitlyTypedAsAny": true}]
	}
}
```

Run:

```
oxlint -c .oxlintrc.json --report-unused-disable-directives repro.ts
```

## Expected

Both ESLint 7.23.0 (with `@typescript-eslint/parser` + `@typescript-eslint/eslint-plugin` 6.21.0) and ESLint 9.39.4 flat config (with `typescript-eslint` 8.62.1) anchor the diagnostic on the arrow function expression — the `=>` token on the `) => {` line (nodeType `ArrowFunctionExpression`, line 5, columns 3–5). The `// eslint-disable-next-line` on line 4 therefore covers the reported line and suppresses the violation; with `--report-unused-disable-directives`, no unused-directive warning is produced and the exit code is 0. oxlint should match this anchoring so the same directive suppresses the finding.

## Actual

oxlint 1.78.0 emits:

```
repro.ts:1:14: error typescript(explicit-module-boundary-types): Missing return type on function help: Define an explicit return type for the function.
repro.ts:4:2: warning: Unused eslint-disable directive (no problems were reported).
```

Exit code 1. The diagnostic is anchored at `1:14` — the `export const showLastActiveUsersWarningDialog =` declaration — not the arrow expression. Because `eslint-disable-next-line` suppression is line-based and the directive on line 4 covers only line 5, it never matches the line-1 diagnostic: the violation stays unsuppressed AND the directive is incorrectly flagged as unused.

## Cross-checked

| Linter | Diagnostic anchor | Directive on line 4 |
| --- | --- | --- |
| ESLint 7.23.0 + @typescript-eslint 6.21.0 | `) =>` line, `ArrowFunctionExpression` | Suppresses; no unused-directive warning; exit 0 |
| ESLint 9.39.4 + typescript-eslint 8.62.1 | `) =>` line (suppressed at 5:3) | Suppresses (shown in `suppressedMessages`); exit 0 |
| oxlint 1.78.0 | 1:14 (`export const` declaration line) | Not matched; reported unused; exit 1 |
